src/debug/Debug.js

```diff
@@ -0,0 +1,112 @@
+    // create debug div
+    const chartDiv = document.createElement('div');
+    chartDiv.id = 'chart-div';
+    chartDiv.style = 'position: absolute; bottom: 0; left: 0; width: 100vw; height: 20rem; background-color: white';
+
+    document.body.appendChild(chartDiv);
+
+    // line graph for nb elements
+    const nbObjectsCanvas = document.createElement('canvas');
+    nbObjectsCanvas.heigth = '20rem';
+    nbObjectsCanvas.width = '40rem';
+    nbObjectsCanvas.id = 'nb-objects';
+    chartDiv.appendChild(nbObjectsCanvas);
+
+    const timestamp = Date.now();
+    const nbObjectsDataset = { label: 'Number of object in Scene', data: [{ x: 0, y: 0 }] };
+    const nbObjectsChartLabel = ['0s'];
+    const nbObjectsChart = new Chart('nb-objects', {
+        type: 'line',
+        data: {
+            labels: nbObjectsChartLabel,
+            datasets: [nbObjectsDataset],
+        },
+        options: {
+            animation: { duration: 10 },
+            scales: {
+                yAxes: [{
+                    display: true,
+                    ticks: {
+                        suggestedMin: 0,    // minimum will be 0, unless there is a lower value.
+                    },
+                }],
+            },
+        },
+    });
+
+    Debug.updateNbObjectsChart = function updateNbObjectsChart() {
+        function countElem(node) {
+            if (!node) {
+                return 0;
+            }
+            let count = 1; // this node
+            if (node.children) {
+                for (const child of node.children) {
+                    count += countElem(child);
+                }
+            }
+            return count;
+        }
+
+        const timeInS = Math.floor((Date.now() - timestamp) / 1000);
+
+        nbObjectsChartLabel.push(`${timeInS}s`);
+        if (nbObjectsChartLabel.length > 20) {
+            nbObjectsChartLabel.shift();
+        }
+        nbObjectsDataset.data.push({ x: timeInS, y: countElem(window.itowns.viewer.scene.gfxEngine.scene3D) });
+        if (nbObjectsDataset.data.length > 20) {
+            nbObjectsDataset.data.shift();
+        }
+        nbObjectsChart.update();
+    };
+
+    const gui = new dat.GUI();
+
+    const state = {
+        showOutline: false,
+        wireframe: false,
+        displayCharts: true,
+    };
+
+    // charts
+    gui.add(state, 'displayCharts').name('Display charts').onChange((newValue) => {
+        if (newValue) {
+            chartDiv.style.display = 'flex';
+        } else {
+            chartDiv.style.display = 'none';
+        }
+    });
+
+    // tiles outline
+    gui.add(state, 'showOutline').name('Show tiles outline').onChange((newValue) => {
+        LayeredMaterial.prototype._showOutline = newValue;
+        var cO = function c0(object) {
+            if (object.materials) {
+                object.materials[0].uniforms.showOutline.value = newValue;
+            }
+        };
+
+        window.itowns.viewer.scene.getMap().tiles.children[0].traverse(cO);
```

Comment:
Ah right, circular dependencies…

So, does `Debug` really need to be exposed in the API? (see comment above about using/enabling it from projects that use iTowns)

And if it does then could the code possibly be split to break the circular dependency? or could the public API be changed? (the only reason we'd need to import `Main` here is because it holds an `ApiGlobe` instance; that instance could be created "elsewhere", with `Main` importing it like it impors `Debug`, and `Debug` then importing that "elsewhere" too; or the public API could be changed to expose the `ApiGlobe` constructor and let users create their own instance of it, and `Debug` would be changed to receive an `ApiGlobe` instance to act on, as well as maybe a container `<div>`).

Feel free to punt those decisions to a later time though (add TODOs in the code or open an issue to track it).